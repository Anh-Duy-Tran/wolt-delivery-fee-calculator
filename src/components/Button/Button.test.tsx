import { expect, it, describe, vi } from 'vitest';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { Button } from '.';

describe('Button Component', () => {
  // Snapshot test
  it('renders correctly', () => {
    const { asFragment } = render(<Button>label</Button>);
    expect(asFragment()).toMatchSnapshot();
  });

  // Test for class application
  it('applies default and custom classes', () => {
    render(<Button className="custom-class" />);
    const button = screen.getByRole('button');
    expect(button).toHaveClass(
      'rounded-[6px] bg-wolt-blue text-white p-3 custom-class'
    );
  });

  // Test for props spreading
  it('spreads additional props to the button element', () => {
    render(<Button aria-label="Test Button" />);
    expect(
      screen.getByRole('button', { name: 'Test Button' })
    ).toBeInTheDocument();
  });

  // Test for handling click events
  it('handles click events', () => {
    const handleClick = vi.fn();
    render(<Button onClick={handleClick} />);
    const button = screen.getByRole('button');
    button.click();
    expect(handleClick).toHaveBeenCalledTimes(1);
  });
});
