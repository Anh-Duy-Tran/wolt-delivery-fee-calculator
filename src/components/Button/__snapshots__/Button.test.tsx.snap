// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component renders correctly 1`] = `
<DocumentFragment>
  <button
    class="rounded-[6px] bg-wolt-blue p-3 text-white"
  >
    label
  </button>
</DocumentFragment>
`;
